src/treebuilders/ConvolutionCalculator.cpp

```diff
@@ -339,16 +339,16 @@ template <int D> void ConvolutionCalculator<D>::tensorApplyOperComp(OperatorStat
         if (oData[i] != nullptr) {
             Eigen::Map<MatrixXd> op(oData[i], os.kp1, os.kp1);
             if (i == D - 1) { // Last dir: Add up into g
-                g += f.transpose() * op;
+                g.noalias() += f.transpose() * op;
```

Comment:
Nice catch. Does this make a difference?

Reply:
I did not test so much, but I think specially the Coulomb potential is significantly faster.

Reply:
Excellent! Have you investigated the effect of the `transpose()`?

Reply:
The transpose is fine: it is a "soft" transpose, i.e. it does not actually move any data.